simexpal/launch/common.py

```diff
@@ -250,16 +265,61 @@ def compile_manifest(run):
 		'timeout': float(exp.info.timeout) if exp.info.timeout is not None else exp.info.timeout,
 		'environ': environ,
 		'output': exp.info.output,
+		'output_extensions': exp.info.output_extensions,
+		'stdout': exp.info.stdout,
 		'workdir': exp.info.workdir
 	})
 
 def invoke_run(manifest):
+
+	def get_qualified_output_file(ext):
+		if ext not in set(manifest.output_extensions + ['out']):
+			raise RuntimeError(
+				f"Unexpected output extension for experiment '{manifest.experiment}': .{ext}\n"
+			)
+		return manifest.output_file_path(ext)
+
+	def substitute_stdout_key(p):
+		if p == 'OUTPUT':
+			return manifest.output_file_path('out')
+		elif p.startswith('OUTPUT:'):
+			return get_qualified_output_file(p.split(':')[1])
+		else:
+			return None
+
+	def stdout_is_valid():
+		for ext in set(manifest.output_extensions + ['out']):
+			if manifest.stdout == manifest.output_file_path(ext):
+				return True
+		return False
+
+	if manifest.output == 'stdout':
+		output_path = manifest.output_file_path('out')
+
+		msg = "Specifying the output path in the 'output' key is deprecated and will be removed in future " \
+				"versions. Use the 'stdout' key instead."
+		warnings.warn(msg, DeprecationWarning)
+	elif manifest.stdout is not None:
+		if manifest.stdout.startswith('@') and manifest.stdout.endswith('@'):
```

Comment:
This check looks strange. Why don't we expand @ variables unconditionally?

Reply:
The ``stdout`` key takes 2 kinds of values.

- an @-variable

or

- some path

We could possibly remove the second case, as I'm not sure that it is practical since it expects a valid path for the exp/variants/rev/instance combination. This means, it will only be valid if the experiment only has one run. For all other runs the path will be invalid. Afterwards, we could expand the @-variable unconditionally like you suggested.

Reply:
Isn't the second case subsumed by just specifying `@OUTPUT@`?

Reply:
Actually, it might be better to just specify the extension of the output file. So `stdout: out` or similar.

Reply:
> Isn't the second case subsumed by just specifying `@OUTPUT@`?

Yes. I will remove this option.

> Actually, it might be better to just specify the extension of the output file. So `stdout: out` or similar.

Ok, sounds good. I will do the change